src/backend/joanie/core/utils/batch_order.py

```diff
@@ -72,8 +72,8 @@ def send_mail_invitation_link(batch_order, invitation_link: str):
                 "title": _(
                     f"{product_title} - A signature is requested for your batch order."
                 ),
-                "email": batch_order.owner.email,
-                "fullname": batch_order.owner.get_full_name(),
+                "email": batch_order.signatory_email,
+                "fullname": f"{batch_order.signatory_firstname} {batch_order.signatory_lastname}",
```

Comment:
As signatory data is optional, maybe we should use owner data as a fallback ?

```suggestion
fullname = batch_order.owner.get_full_name()
if batch_order.signatory_firstname or batch_order.signatory_lastname:
     fullname = f"{batch_order.signatory_firstname} {batch_order.signatory_lastname}"
     
     […]
                "email": batch_order.signatory_email or batch_order.owner.email,
                "fullname": fullname,
```

Reply:
Okay, I will need to update as well the preparation of the signature workflow if signatory value may not always be present.

Reply:
Signatory data is not optional through our API though...